### PR TITLE
fix: Adjacent JSX elements should be allowed in mdx

### LINKED ISSUE #6331: mdx: SyntaxError: Adjacent JSX elements must be wrapped in an enclosing tag.
**Prettier 1.18.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAgDMBnInCdfAchKoB0oNs99h8AFHOAYQgBM4+AL5kK1AHTiA9JCzQEMYvSgMAPMThgYqaAD4G+fKs49+gyAIC8wEuK4BDPgDFUAGzgBlAJ5QwACgB9AL5UHCh7dEEAamopdwBzezAvACZxRSoAShFXeyh4gFd7eLhLOhAYL0w4YjAcVEwYct1VKRNeAX0oVo0tHSgu9U1tPQMjEIA3fDBc4mIAOQjS8sIcYsjYIkc4AFoIAqaQfGJK92s2SFdcJGouPiphIS7DQwB1AAsvAH58AEEoLz4EKEQhwLi+OBfKFfMatSaDKS9EYDEAAGhAEEa-WIyFA9hw5AA7ux8QgcSh7BMIKg+GiQAAjNZgADWcBgHkwSVQ+WQMBwBTg6PeMHQrg+qHgxE5YE8ZIlqAmEq8yHAczp3I0eE463syEI9lcGnRACtiAAPABCTNZ7KWABluXA9QajSBTWaPNz4u4AIoFCDwZ2GwUgTk4TUq9B8M10zD1WCvGkwd7IAAcAAZ0XGIBpXmtMCq4zUwRMneiAI7++CcTHkkD2Yg7KBwOACWnoriV0JwbXxdC6pD64PojToVC8-kh4he31Vp2Dl0hmD2emJvjJ5ApdF8+xuL28dD9lVQeR0goaAAqK-JQ40QiEQA)
```sh
# Options (if any):
--parser mdx
```

**Input:**
```mdx
import fs from 'fs'
import { PreCode } from '../components'

<section>
  <PreCode code={fs.readFileSync(__dirname + '/legacy2.ts')} language="typescript"></PreCode>
</section>
<section>
  <div className="fragment fade-out" style={{ color: 'red' }}>
    Why? Any difference????
  </div>
</section>
```

**Output:**
```mdx
SyntaxError: Adjacent JSX elements must be wrapped in an enclosing tag. Did you want a JSX fragment <>...</>? (4:1)
  2 |   <PreCode code={fs.readFileSync(__dirname + '/legacy2.ts')} language="typescript"></PreCode>
  3 | </section>
> 4 | <section>
    | ^
  5 |   <div className="fragment fade-out" style={{ color: 'red' }}>
  6 |     Why? Any difference????
  7 |   </div>
```

**Expected behavior:**
No Error

### PR DESCRIPTION
**Pull request by JounQin**
fix #6331

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by JounQin** [comment]
@vjeux @ikatyang Hey guys, it's my first PR for `prettier`, would you like to help me to make it better?

image

File: src/language-markdown/parser-markdown.js [hunk 1]
```diff
@@ -59,7 +63,23 @@ function htmlToJsx() {
...
+      );
+
+      // find out if there are adjacent JSX elements which should be allowed in mdx alike in markdown
+      if (htmlNodes.length <= 1) {
+        return Object.assign({}, node, { type: "jsx" });
+      }
+
+      return htmlNodes.map(htmlNode => ({
+        type: "jsx",
+        value: printDocToString(
```
Comment (JounQin) [inline src/language-markdown/parser-markdown.js]: Anyone knows how to get the original text from ast `htmlNode`? Or is a better way to split the elements text?

**Comment by JounQin** [comment]
May be blocked by #6336.

File: src/language-markdown/parser-markdown.js (hunk 1 above)
Comment (JounQin) [reply src/language-markdown/parser-markdown.js]: OK, I finally find out that it's so easy to get the original text from ast position. 😂

**Comment by JounQin** [comment]
I don't know why my last commit 8b74f7be48cf883a0d6b0f2b9e3c74ef0dd5c175 get wrong, so I'm going to split it into another PR in case not blocking current one.

**Comment by lipis** [comment]
Fix linting :)

**Comment by JounQin** [comment]
@lipis done.

**Comment by lipis** [comment]
🍺

**Pull request merged**